[CI] SmokeTestMultiNodeClientYamlTestSuiteIT test {yaml=indices.create/40_dynamic_strings_auto_keyword/Dynamic string with auto text later disabled omits text subfield} failing

**Build Scans:**
- [elasticsearch-intake #39894 / part6](https://gradle-enterprise.elastic.co/s/q3roz3zb3zad2)
- [elasticsearch-pull-request #146771 / part-6](https://gradle-enterprise.elastic.co/s/oxuduermbzpg4)

**Reproduction Line:**
```
./gradlew ":qa:smoke-test-multinode:yamlRestTest" --tests "org.elasticsearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT" -Dtests.method="test {yaml=indices.create/40_dynamic_strings_auto_keyword/Dynamic string with auto text later disabled omits text subfield}" -Dtests.seed=18118F651A143001 -Dtests.locale=en-NU -Dtests.timezone=Asia/Taipei -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Byaml%3Dindices.create%2F40_dynamic_strings_auto_keyword%2FDynamic%20string%20with%20auto%20text%20later%20disabled%20omits%20text%20subfield%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: org.junit.internal.runners.model.MultipleFailureException: There were 2 errors:
  java.lang.AssertionError(field [test_dynamic_strings.mappings.properties.new_field.type] is null)
  java.lang.AssertionError(field [test_dynamic_strings.mappings.properties.new_field.fields.keyword.type] is null)
```

**Issue Reasons:**
- [main] 1 consecutive failures in test test {yaml=indices.create/40_dynamic_strings_auto_keyword/Dynamic string with auto text later disabled omits text subfield}
- [main] 1 consecutive failures in step part6
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 2 failures in test test {yaml=indices.create/40_dynamic_strings_auto_keyword/Dynamic string with auto text later disabled omits text subfield} (4.3% fail rate in 47 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -560,9 +560,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.csv.CsvFormatSpecIT
   method: test {csv-spec:csv-union-by-name.multiFileUnionByNameRecipeWithPruning [LOCAL]}
   issue: https://github.com/elastic/elasticsearch/issues/149352
-- class: org.elasticsearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT
-  method: test {yaml=indices.create/40_dynamic_strings_auto_keyword/Dynamic string with auto text later disabled omits text subfield}
-  issue: https://github.com/elastic/elasticsearch/issues/149372
 - class: org.elasticsearch.index.mapper.DynamicMappingIT
   method: testDynamicStringMappingWithoutAutoTextSubfield
   issue: https://github.com/elastic/elasticsearch/issues/149374

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.create/40_dynamic_strings_auto_keyword.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.create/40_dynamic_strings_auto_keyword.yml
@@ -27,6 +27,10 @@ setup:
   - match: { test_dynamic_strings_default.defaults.index.mapping.dynamic_strings.auto_text: "true" }
 
   - do:
+      cluster.health:
+        wait_for_events: languid
+
+  - do:
       indices.get_mapping:
         index: test_dynamic_strings_default
 
@@ -55,6 +59,10 @@ setup:
         index: test_dynamic_strings_no_text
 
   - match: { test_dynamic_strings_no_text.settings.index.mapping.dynamic_strings.auto_text: "false" }
+
+  - do:
+      cluster.health:
+        wait_for_events: languid
 
   - do:
       indices.get_mapping:
@@ -88,6 +96,10 @@ setup:
   - match: { test_dynamic_strings.settings.index.mapping.dynamic_strings.auto_text: "true" }
 
   - do:
+      cluster.health:
+        wait_for_events: languid
+
+  - do:
       indices.get_mapping:
         index: test_dynamic_strings
 
@@ -112,6 +124,10 @@ setup:
         index: test_dynamic_strings
 
   - match: { test_dynamic_strings.settings.index.mapping.dynamic_strings.auto_text: "false" }
+
+  - do:
+      cluster.health:
+        wait_for_events: languid
 
   - do:
       indices.get_mapping:

